libglusterfs/src/common-utils.c

```diff
@@ -3682,22 +3699,38 @@ gf_is_same_address(char *name1, char *name2)
     gf_boolean_t ret = _gf_false;
     int gai_err = 0;
     struct addrinfo hints;
+    xlator_t *this = THIS;
+    glusterfs_ctx_t *ctx = this->ctx;
+    gf_boolean_t flag = _gf_false;
+
+    if (ctx && ctx->dnscache)
+        flag = _gf_true;
 
     memset(&hints, 0, sizeof(hints));
     hints.ai_family = AF_UNSPEC;
 
-    gai_err = getaddrinfo(name1, NULL, &hints, &addr1);
-    if (gai_err != 0) {
-        gf_smsg(name1, GF_LOG_WARNING, 0, LG_MSG_GETADDRINFO_FAILED, "error=%s",
-                gai_strerror(gai_err), NULL);
-        goto out;
-    }
+    if (flag) {
+        addr1 = gf_dns_lookup_address_cached(name1, ctx->dnscache);
+        if (!addr1)
+            return ret;
 
-    gai_err = getaddrinfo(name2, NULL, &hints, &addr2);
-    if (gai_err != 0) {
-        gf_smsg(name2, GF_LOG_WARNING, 0, LG_MSG_GETADDRINFO_FAILED, "error=%s",
-                gai_strerror(gai_err), NULL);
-        goto out;
+        addr2 = gf_dns_lookup_address_cached(name2, ctx->dnscache);
+        if (!addr2)
+            return ret;
+    } else {
+        gai_err = getaddrinfo(name1, NULL, &hints, &addr1);
+        if (gai_err != 0) {
+            gf_smsg(name1, GF_LOG_WARNING, 0, LG_MSG_GETADDRINFO_FAILED,
+                    "error=%s", gai_strerror(gai_err), NULL);
+            goto out;
+        }
+
+        gai_err = getaddrinfo(name2, NULL, &hints, &addr2);
+        if (gai_err != 0) {
+            gf_smsg(name2, GF_LOG_WARNING, 0, LG_MSG_GETADDRINFO_FAILED,
+                    "error=%s", gai_strerror(gai_err), NULL);
+            goto out;
+        }
```

Comment:
This code is duplicated, and it's the same used in `gf_dns_lookup_address_cached()`. It would be better to move it to a new function and reuse it.

Reply:
done